lib/features/digital_guide_view/presentation/digital_guide_view.dart

```diff
@@ -0,0 +1,118 @@
+import "../../../utils/context_extensions.dart";
+import "../../../utils/determine_contact_icon.dart";
+import "../../../utils/phone_numbers.dart";
+import "../../../widgets/detail_views/contact_section.dart";
+import "../../../widgets/detail_views/detail_view_app_bar.dart";
+import "../../../widgets/report_change_button.dart";
+import "../data/models/digital_guide_response.dart";
+import "../data/repository/digital_guide_repository.dart";
+import "../data/repository/image_repository.dart";
+import "widgets/accessibility_button.dart";
+import "widgets/digital_guide_data_source_link.dart";
+import "widgets/digital_guide_features_section.dart";
+import "widgets/headlines_section.dart";
+
+@RoutePage()
+class DigitalGuideView extends ConsumerWidget {
+  const DigitalGuideView({
+    @PathParam("id") required this.id,
+  });
+
+  final int id;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final asyncDigitalGuideData = ref.watch(getDigitalGuideDataProvider(id));
+    // question: Should the app bar appear during loading or when there's an error?
+    // Now it doesn't, neither does it appear on SKS menu screen
+    return asyncDigitalGuideData.when(
+      data: _DigitalGuideView.new,
+      // To-do nice error
+      error: (error, stackTrace) => Text("API error occured: $error"),
+      loading: () => const Scaffold(
+        body: Center(
+          child: CircularProgressIndicator(),
+        ),
+      ),
+    );
+  }
+}
+
+class _DigitalGuideView extends ConsumerWidget {
+  const _DigitalGuideView(this.digitalGuideResponse);
+
+  final DigitalGuideResponse digitalGuideResponse;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final asyncImageResponse =
+        ref.watch(getImageDataProvider(digitalGuideResponse.images[0]));
+
+    return Scaffold(
+      appBar: DetailViewAppBar(
+        title: context.localize.map,
+        actions: [
+          AccessibilityButton(),
+        ],
+      ),
+      body: CustomScrollView(
+        slivers: [
+          SliverList(
+            delegate: SliverChildListDelegate([
+              const SizedBox(height: DigitalGuideConfig.heightSmall),
+              asyncImageResponse.when(
+                data: (imageResponseData) => Image.network(
+                  imageResponseData.imageUrl,
+                  fit: BoxFit.fitWidth,
+                ),
+                loading: () => const Center(child: CircularProgressIndicator()),
+                // To-do nice error
+                error: (error, stack) => Text("Error loading images: $error"),
+              ),
+              HeadlinesSection(
+                name: digitalGuideResponse.translations.pl.name,
+                description: digitalGuideResponse.translations.pl.extendedName,
+              ),
+              ContactSection(
+                list: IList<ContactIconsModel>([
+                  ContactIconsModel(
+                    text: digitalGuideResponse.translations.pl.address
```

Comment:
Probably we should make it as an extension. And what about versions in other languages? We should take that into consideration as well

Reply:
This one is tricky because there is only onle translation in API - Polish. Moreover, they have json structured this way

Reply:
Hmm, till it's only in polish we can keep that. Please add comment regarding that situation. Later on we will need to think about how to solve this problem.

Reply:
For now, we can keep that in the data layer. Ui doesn;'t need to know which language user uses. Later on, when on API site other languages will be available we will need to add some kind of logic to determine it.